pkg/workspace/library_loader.go

```diff
@@ -0,0 +1,190 @@
+package workspace
+
+import (
+	"fmt"
+
+	"github.com/k14s/ytt/pkg/files"
+	"github.com/k14s/ytt/pkg/texttemplate"
+	"github.com/k14s/ytt/pkg/yamlmeta"
+	"github.com/k14s/ytt/pkg/yamltemplate"
+	"github.com/k14s/ytt/pkg/yttlibrary"
+)
+
+type LibraryLoader struct {
+	*Library
+	UI     files.UI
+	Opts   TemplateLoaderOpts
+	Values interface{}
+}
+
+type EvalResult struct {
+	DocSet      *yamlmeta.DocumentSet
+	DocSets     map[*FileInLibrary]*yamlmeta.DocumentSet
+	OutputFiles []files.OutputFile
+}
+
+func NewLibraryLoader(lib *Library, parentValuesAst interface{}, ui files.UI, loaderOpts TemplateLoaderOpts) *LibraryLoader {
+	return &LibraryLoader{
+		UI:      ui,
+		Opts:    loaderOpts,
+		Library: lib,
+		Values:  parentValuesAst,
+	}
+}
+
+func (ll *LibraryLoader) LoadValues() error {
+	valuesFiles, err := ll.loadValueFiles(ll.Library, true)
+	if err != nil {
+		return err
+	}
+
+	noValuesLoader := NewTemplateLoader(nil, ll.UI, ll.Opts)
+
+	ll.Values, err = DataValuesPreProcessing{valuesFiles, ll.Values, noValuesLoader, ll.Opts.IgnoreUnknownComments}.Apply()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (ll *LibraryLoader) loadValueFiles(lib *Library, recurse bool) ([]*FileInLibrary, error) {
+	var valuesFiles []*FileInLibrary
+
+	for _, file := range lib.files {
```

Comment:
we shouldnt be accessing Library private methods. arguably we should probably create a separate package instead of moving this into workspace.

Reply:
> arguably we should probably create a separate package instead of moving this into workspace.
i take this back. but we should continue to use ListAccessibleFiles() here.

Reply:
I specifically avoided `ListAccessibleFiles()` because I was thinking of scoping the value files relative to their directory. I mentioned it here: https://github.com/k14s/ytt/pull/24#issuecomment-503953521

The idea was that value files in subdirectories would only affect `data.values` of their directory and subdirectories but not their parent directory.

This is however not implemented in this PR which is just a refactoring, but I prepared the work here. I can remove it from this PR though and move it to a further PR.

Reply:
> The idea was that value files in subdirectories would only affect data.values of their directory and subdirectories but not their parent directory.

im not really on board with this idea because `data.values` via data library should be returning same values throughout same library. i also dont see a compelling use case for separating data values per directory (since directory isnt a first class concept, library is though).

Reply:
Library objects are split per directory, that's what I meant. Currently, there is no real separation between a Library and its child libraries, but we can introduce value files that would not affect the parent libraries as they do currently.